catalyst/contrib/data/cv/datasets/image.py

```diff
@@ -0,0 +1,197 @@
+from typing import Callable, Optional, Tuple, Union
+from collections import OrderedDict
+import math
+from pathlib import Path
+
+import cv2
+import skimage.io
+
+import torch
+from torch.utils.data import Dataset
+
+
+def get_image_margins(
+    image_h: int,
+    image_w: int,
+    tile_size_h: int,
+    tile_size_w: int,
+    tile_step_h: int,
+    tile_step_w: int,
+):
+    """
+    Compute image margins to fit tile parameters.
+
+    Args:
+        image_h: image height
+        image_w: image width
+        tile_size_h: tile size in height
+        tile_size_w: tile size in width
+        tile_step_h: tile step in height
+        tile_step_w: tile step in width
+
+    Returns:
+        OrderedDict with margins at all 4 edges of the image.
+    """
+    overlap_h = tile_size_h - tile_step_h
+    overlap_w = tile_size_w - tile_step_w
+
+    num_tiles_h = math.ceil((image_h - overlap_h) / tile_step_h)
+    num_tiles_w = math.ceil((image_w - overlap_w) / tile_step_w)
+
+    num_extra_pixels_h = tile_step_h * num_tiles_h + overlap_h - image_h
+    num_extra_pixels_w = tile_step_w * num_tiles_w + overlap_w - image_w
+
+    margin_bottom = math.floor(num_extra_pixels_h / 2)
+    margin_top = math.ceil(num_extra_pixels_h / 2)
+    margin_left = math.floor(num_extra_pixels_w / 2)
+    margin_right = math.floor(num_extra_pixels_w / 2)
+
+    margins = OrderedDict(
+        margin_bottom=margin_bottom,
+        margin_top=margin_top,
+        margin_left=margin_left,
+        margin_right=margin_right,
+    )
```

Comment:
Why `OrderedDict` instead of `Dict` or `NamedTuple` or `DataClass`?
It is used as usual dict below.